docs/semantic_catalog/quickstart.md

```diff
@@ -0,0 +1,344 @@
+interact with data and database systems using everyday language in the place of complex SQL syntax.
+
+## How Text-to-SQL works
+
+LLMs are capable of writing valid SQL queries without further training on the SQL language. However, knowing SQL is not
+enough to author queries that accurately answer business questions. Humans and LLMs need to:
+
+1. Know the SQL language
+2. Understand the database schema
+3. Understand the format and meaning of the data in the database
+
+LLMs know the language. The postgres catalog contains the definitions of the schema and data. The semantic catalog acts
+as a bridge between the structural definitions and a human-level understanding in natural language.
+
+One could dump the entire schema and data to the LLM prompt. For very small databases, this might work. As schemas get
+larger, this approach will consume too many tokens, too much time, and risks leading the LLM astray with irrelevant details.
+
+For demanding applications, we use the semantic catalog to include only the schema and sample data that is *relevant* to
+the question posed. This light-touch solution reduces the number of tokens consumed and avoids introducing irrelevant
+details to the LLM.
+
+### The semantic catalog
+
+A semantic catalog is a comprehensive knowledge repository that bridges the gap between natural language and database
+structures. The term _semantic_ refers to the meaning or interpretation of language and symbols, focusing on the
+relationships between words, phrases, and concepts rather than just their literal definitions. The semantic catalog is
+a layer that enriches raw database objects with meaningful context that natural language processing models can leverage.
+
+The semantic catalog consists of the following components:
+
+- **Database objects**: with natural language descriptions that provide human-readable context for technical elements such
+   as tables, columns, and functions. This mapping helps the system understand what users mean when they reference
+   business concepts rather than technical database terms.
+- **SQL examples**: paired with natural language descriptions that demonstrate how specific questions translate into query
+   structures.
+- **Facts**: expressed as natural language statements about the dataset/schema, facts provide additional domain knowledge
+   that might not be explicitly encoded in the database structure but is crucial for understanding user intent.
+
+By embedding these components, your semantic catalog enables vector similarity searches that can retrieve relevant
+context when processing new queries. This retrieval-augmented generation (RAG) approach ensures that when constructing
+SQL queries from natural language inputs, Text-to-SQL has access to the most pertinent information about your database
+structure, usage patterns, and domain knowledge.
+
+### Semantic Catalog Features
```

Comment:
I'm not sure that these features are the headline features of the semantic catalog. What about:
- automatically creating object descriptions
- importing object descriptions
- managing object descriptions